[BUG] Redshift Serverless SDK missing in connector athena-redshift

**Describe the bug**
When using Redshift Serverless features (for example, using a JDBC connection string such as redshift://jdbc:redshift:iam://test.123456.us-east-1.redshift-serverless.amazonaws.com:5439/dev), the connector will throw an error

```
com/amazonaws/services/redshiftserverless/AWSRedshiftServerlessClientBuilder: java.lang.NoClassDefFoundError
java.lang.NoClassDefFoundError: com/amazonaws/services/redshiftserverless/AWSRedshiftServerlessClientBuilder
	at com.amazon.redshift.core.IamHelper.setClusterCredentials(IamHelper.java:645)
	at com.amazon.redshift.core.IamHelper.setIAMCredentials(IamHelper.java:600)
	at com.amazon.redshift.core.IamHelper.setIAMProperties(IamHelper.java:360)
	at com.amazon.redshift.jdbc.RedshiftConnectionImpl.<init>(RedshiftConnectionImpl.java:284)
	at com.amazon.redshift.Driver.makeConnection(Driver.java:481)
	at com.amazon.redshift.Driver.connect(Driver.java:302)
	at java.sql/java.sql.DriverManager.getConnection(Unknown Source)
	at java.sql/java.sql.DriverManager.getConnection(Unknown Source)
	at com.amazonaws.athena.connectors.jdbc.connection.GenericJdbcConnectionFactory.getConnection(GenericJdbcConnectionFactory.java:90)
	at com.amazonaws.athena.connectors.jdbc.manager.JdbcMetadataHandler.doListSchemaNames(JdbcMetadataHandler.java:148)
	at com.amazonaws.athena.connector.lambda.handlers.MetadataHandler.doHandleRequest(MetadataHandler.java:259)
	at com.amazonaws.athena.connector.lambda.handlers.CompositeHandler.handleRequest(CompositeHandler.java:145)
	at com.amazonaws.athena.connector.lambda.handlers.CompositeHandler.handleRequest(CompositeHandler.java:113)
Caused by: java.lang.ClassNotFoundException: com.amazonaws.services.redshiftserverless.AWSRedshiftServerlessClientBuilder. Current classpath: file:/var/task/
```

This is due to the pom.xml missing the aws-java-sdk-redshiftserverless dependency being needed. Adding

```
<dependency>
    <groupId>com.amazonaws</groupId>
    <artifactId>aws-java-sdk-redshiftserverless</artifactId>
    <version>${aws-sdk.version}</version>
</dependency>
```

to athena-redshift/pom.xml fixes the issue.

**To Reproduce**
Steps to reproduce the behavior:
1. Create a Redshift Serverless DB.
2. In Athena, use a Redshift data source connector with a connection string similar to "redshift://jdbc:redshift:iam://test.123456.us-east-1.redshift-serverless.amazonaws.com:5439/dev".
3. The Athena console will show an error

> Error fetching Associated databases
> Failed to invoke lambda function due to com.amazonaws.services.lambda.invoke.LambdaFunctionException: com/amazonaws/services/redshiftserverless/AWSRedshiftServerlessClientBuilder

and the CloudWatch logs will show the error given above.

**Expected behavior**
No error

**Screenshots / Exceptions / Errors**
See above

**Connector Details (please complete the following information):**
 - Version: 2024.15.2
 - Name: redshift
 - Athena Query IDs: None

**Additional context**
None

--- a/athena-redshift/pom.xml
+++ b/athena-redshift/pom.xml
@@ -38,6 +38,11 @@
             <artifactId>aws-java-sdk-redshift</artifactId>
             <version>${aws-sdk.version}</version>
         </dependency>
+        <dependency>
+            <groupId>com.amazonaws</groupId>
+            <artifactId>aws-java-sdk-redshiftserverless</artifactId>
+            <version>${aws-sdk.version}</version>
+        </dependency>
         <!-- https://mvnrepository.com/artifact/software.amazon.awscdk/redshift -->
         <dependency>
             <groupId>software.amazon.awscdk</groupId>

